lib/seek/samples/sample_metadata_updater.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Seek
+  module Samples
+
+    class SampleMetadataUpdateException < StandardError; end
+
+    # Class to handle the updating of sample metadata after updating Sample type
+    class SampleMetadataUpdater
+      def initialize(sample_type, attribute_changes, user)
+        @sample_type = sample_type
+        @attribute_change_maps = attribute_changes
+        @user = user
+      end
+
+      def update_metadata
+        return if @attribute_change_maps.blank? || @sample_type.samples.blank? || @sample_type.nil? || @user.nil?
+
+        User.with_current_user(@user) do
+          @sample_type.samples.each do |sample|
```

Comment:
I think this may have to work in batches, as updating thousands of samples could take a long time and the job may time out (as has happened with deletes)